src/aws-cpp-sdk-core/include/smithy/Smithy_EXPORTS.h

```diff
@@ -0,0 +1,28 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#pragma once
+
+#ifdef _MSC_VER
+//disable windows complaining about max template size.
+    #pragma warning (disable : 4503)
+#endif // _MSC_VER
+
+#if defined (USE_WINDOWS_DLL_SEMANTICS) || defined (_WIN32)
```

Comment:
We decided to make smithy a different set of exports in the event that it would be split off into its own library, and we need the win annotations for creating the lib. It [largely copies the export model that we already have for core](https://github.com/aws/aws-sdk-cpp/blob/main/src/aws-cpp-sdk-core/include/aws/core/Core_EXPORTS.h#L10-L32).

Reply:
talked and since these are new exports we will align with how [CRT does visibility on linux](https://github.com/awslabs/aws-c-common/blob/9f36d07189435db27f746c10e39ba58511fadd0d/include/aws/common/exports.h#L20-L24)